### PR TITLE
Move cURL calls to a seperate Request library 

### PR DESCRIPTION
**Pull request by TomAbel**
#124

for OpenCart 1.5.6.4

realized this:
```
} catch (SmailyError $error) {
  $msg = $error->getMessage();
  $this->log->write($msg);
  echo($msg);
```
voids the purpose of `__toString()` in BaseError class. Having `$error->getMessage() ` would just print out _GET request to Smaily API failed_ which isn't very helpful. Modified it so we're writing and echoing the $error variable, properly formatted.

### CODE REVIEW

File: upload/admin/model/smailyforopencart/admin.php [hunk 1]
```diff
@@ -22,13 +22,57 @@ public function editSettingValue($group = '', $key = '', $value = '', $store_id
...
+     * @return void
+     */
+    public function saveAPICredentials($subdomain, $username, $password) {
+        $settings = [
+            'subdomain' => $this->db->escape($subdomain),
+            'password' => $this->db->escape($password),
+            'username' => $this->db->escape($username),
+        ];
+        $this->editSettingValue('smaily', 'smaily_api_credentials', $settings);
+
```
Comment (kaittodesk) [inline upload/admin/model/smailyforopencart/admin.php:75]: Formatting: excess empty line.

File: upload/admin/model/smailyforopencart/admin.php [hunk 2]
```diff
@@ -22,13 +22,57 @@ public function editSettingValue($group = '', $key = '', $value = '', $store_id
...
+    }
+
```
Comment (kaittodesk) [inline upload/admin/model/smailyforopencart/admin.php:77]: Formatting: excess empty line.

**Review by kaittodesk — CHANGES_REQUESTED** [review]

File: upload/admin/model/smailyforopencart/admin.php (hunk 1 above)
Comment (TomAbel) [reply upload/admin/model/smailyforopencart/admin.php:75]: Removed in 4fe0aac

File: upload/admin/model/smailyforopencart/admin.php (hunk 2 above)
Comment (TomAbel) [reply upload/admin/model/smailyforopencart/admin.php:77]: Removed in 4fe0aac

**Review by kaittodesk — APPROVED** [review]

**Pull request merged**